shell/platform/tizen/BUILD.gn

```diff
@@ -87,41 +53,169 @@ source_set("flutter_tizen") {
     "$custom_sysroot/usr/include/eo-1",
     "$custom_sysroot/usr/include/evas-1",
     "$custom_sysroot/usr/include/system",
-    "$custom_sysroot/usr/include/wayland-extension"
+    "$custom_sysroot/usr/include/wayland-extension",
+    # For Evas_GL.
+    "$custom_sysroot/usr/include/ecore-con-1",
+    "$custom_sysroot/usr/include/ecore-file-1",
+    "$custom_sysroot/usr/include/edje-1",
+    "$custom_sysroot/usr/include/eet-1",
+    "$custom_sysroot/usr/include/efl-1/interfaces",
+    "$custom_sysroot/usr/include/efreet-1",
+    "$custom_sysroot/usr/include/elementary-1",
+    "$custom_sysroot/usr/include/ethumb-1",
+    "$custom_sysroot/usr/include/ethumb-client-1",
   ]
-  
-  lib_dirs = [ root_out_dir, "$custom_sysroot/usr/lib" ]
 
-  cflags_cc = [
-    "-Wno-newline-eof",
-    "-Wno-macro-redefined",
-  ]
+  lib_dirs = [ "$custom_sysroot/usr/lib" ]
+}
 
-  libs = [
-    "base-utils-i18n",
-    "capi-appfw-application",
-    "capi-system-info",
-    "capi-system-system-settings",
-    "dlog",
-    "ecore",
-    "ecore_imf",
-    "ecore_input",
-    "eina",
-    "EGL",
-    "evas",
-    "flutter_engine",
-    "GLESv2",
-    "tbm",
-    "tdm-client",
-    "wayland-client",
-  ]
+# Template for the embedder build. Used to generate embedders for different
+# device profiles. The output library name is "flutter_tizen_[profile]".
+#
+# If use_evas_gl_renderer is provided as true, the Evas_GL renderer is used,
+# otherwise the Ecore_Wl2 renderer is used.
+template("embedder_for_profile") {
+  forward_variables_from(invoker, [ "use_evas_gl_renderer" ])
 
-  if (tizen_sdk_4) {
-    sources += [ "tizen_renderer_ecore_wl.cc" ]
-    libs += [ "ecore_wayland", "wayland-egl" ]
-    defines = [ "FLUTTER_TIZEN_4" ]
-  } else {
-    sources += [ "tizen_renderer_ecore_wl2.cc" ]
-    libs += [ "ecore_wl2" ]
+  if (!defined(use_evas_gl_renderer)) {
+    use_evas_gl_renderer = false
   }
+
+  shared_library("flutter_tizen_${target_name}") {
```

Comment:
I've found something confusing (at least for gn novice): the value of `target_name` within the body of `shared_library`, i.e. between lines 85 and 164, is changed from `mobile`/`werable`/`tv`/`common` to `flutter_tizen_mobile`/`flutter_tizen_werable`/`flutter_tizen_tv`/`flutter_tizen_common` respectively.

I'm not sure, if it's an issue, but it causes "undefined reference" errors when using code similar to the one from the diff from this comment https://github.com/flutter-tizen/engine/pull/65#issuecomment-827318857 to extend `libs`, i.e.: 

```
+    if (target_name == "mobile") { # should be target_name == "flutter_tizen_mobile"
+      libs += [ "cbhm" ]
+    }
```

I've printed the `target_name` values before, within and after `shared_library()` call (using this code: https://github.com/pwasowski2/engine/commit/0a41130c75706da6bddc546f07d4e64497c0cd37) and here's what gn uses as `target_name`s:
```
target_name before 'shared_library()': mobile
target_name in 'shared_library()': flutter_tizen_mobile
target_name == "mobile" is FALSE
target_name == "flutter_tizen_mobile" is TRUE
target_name after 'shared_library()': mobile
...
```

@swift-kim, is this change of `target_name` value desired?

Reply:
@pwasowski2 You're right. My comment was misleading because I didn't test the change suggested in the comment.

You can either:

- Use `invoker.target_name` (instead of `target_name`) inside `shared_library()`
- Define a new variable (e.g. `profile = target_name`) outside `shared_library()` and use that inside `shared_library()`

Actually there is a caveat about this in the GN documentation: https://gn.googlesource.com/gn/+/master/docs/reference.md#var_target_name

Reply:
Thanks for the explanation :)